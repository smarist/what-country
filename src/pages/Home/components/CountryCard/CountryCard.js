import { Box, Typography } from '@material-ui/core';
import React from 'react';
import { Link } from 'react-router-dom';
import useStyles from './CountryCard.style';

function CountryCard({
  population,
  region,
  capital,
  flag,
  name,
}) {
  const classes = useStyles();
  return (
    <Box className={classes.cardBox}>
      <Box className={classes.flagImage}>
        <Link to={`/${name}`} key={name}>
          <img src={flag} alt="flag" className={classes.flag} />
        </Link>
      </Box>
      <Box className={classes.cardDetails}>
        <Typography className={classes.countryName}>{name}</Typography>
        <Box className={classes.textBox}>
          <Typography className={classes.cardTitle}>Population:</Typography>
          <Typography className={classes.cardText}>{population}</Typography>
        </Box>
        <Box className={classes.textBox}>
          <Typography className={classes.cardTitle}>Region:</Typography>
          <Typography className={classes.cardText}>{region}</Typography>
        </Box>
        <Box className={classes.textBox}>
          <Typography className={classes.cardTitle}>Capital:</Typography>
          <Typography className={classes.cardText}>{capital}</Typography>
        </Box>
      </Box>
    </Box>
  );
}

export default CountryCard;
