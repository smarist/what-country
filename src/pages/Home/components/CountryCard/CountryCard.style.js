import { makeStyles } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  cardBox: {
    // width: '240px',
    // height: '280px',
    boxShadow: theme.palette.boxShadow.main,
    borderRadius: '3px',
    backgroundColor: theme.palette.bgColor2.main,
  },
  flagImage: {
    height: '150px',
    width: '100%',
    borderRadius: '3px 3px 0 0',
  },
  flag: {
    width: '100%',
    height: '100%',
    objectFit: 'cover',
  },
  cardDetails: {
    padding: '20px',
  },
  textBox: {
    display: 'flex',
    marginTop: '5px',
  },
  countryName: {
    fontSize: '16px !important',
    fontWeight: '600',
    marginBottom: '15px',
  },
  cardTitle: {
    fontSize: '14px !important',
    fontWeight: '600',
    color: theme.palette.textColor2.main,
  },
  cardText: {
    fontSize: '14px !important',
    marginLeft: '4px !important',
    color: theme.palette.textColor3.main,
  },
}));

export default useStyles;
