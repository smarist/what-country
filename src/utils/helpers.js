export function objectToArray(val) {
  const obj = val;
  return Object.entries(obj);
}

let timeoutId;

export function debounceHandler(func, delay) {
  if (timeoutId) clearTimeout(timeoutId);
  timeoutId = setTimeout(func, delay);
}
