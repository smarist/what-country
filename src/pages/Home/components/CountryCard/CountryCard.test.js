import React from 'react';
import { render } from '@testing-library/react';
import CountryCard from './CountryCard';
import { withThemeAndStore } from '../../../../tests/helpers';

jest.mock('react-router-dom', () => ({
  useParams: jest.fn().mockReturnValue({ nameId: '' }),
  useNavigate: jest
    .fn()
    .mockReturnValue({ goBack: jest.fn() }),
  useLocation: jest.fn().mockReturnValue({}),
  Link: (propVal) => {
    const LinkNav = 'link-navigation';
    return <LinkNav {...propVal} />;
  },
}));

const props = {
  population: '',
  region: '',
  capital: '',
  flag: '',
  name: '',
};

describe('Testing Country Card Component', () => {
  it('Should render successfully', () => {
    const { queryByTestId } = render(withThemeAndStore(CountryCard, props));

    const countryCard = queryByTestId('country-card');
    expect(countryCard).toBeTruthy();
    expect(countryCard).toMatchSnapshot();
  });
});
