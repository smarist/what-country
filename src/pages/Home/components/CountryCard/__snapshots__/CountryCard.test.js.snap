// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Country Card Component Should render successfully 1`] = `
<div
  class="MuiBox-root MuiBox-root-9 makeStyles-cardBox-1"
  data-testid="country-card"
>
  <div
    class="MuiBox-root MuiBox-root-10 makeStyles-flagImage-2"
  >
    <link-navigation
      to="/"
    >
      <img
        alt="flag"
        class="makeStyles-flag-3"
        src=""
      />
    </link-navigation>
  </div>
  <div
    class="MuiBox-root MuiBox-root-11 makeStyles-cardDetails-4"
  >
    <p
      class="MuiTypography-root makeStyles-countryName-6 MuiTypography-body1"
    />
    <div
      class="MuiBox-root MuiBox-root-12 makeStyles-textBox-5"
    >
      <p
        class="MuiTypography-root makeStyles-cardTitle-7 MuiTypography-body1"
      >
        Population:
      </p>
      <p
        class="MuiTypography-root makeStyles-cardText-8 MuiTypography-body1"
      />
    </div>
    <div
      class="MuiBox-root MuiBox-root-13 makeStyles-textBox-5"
    >
      <p
        class="MuiTypography-root makeStyles-cardTitle-7 MuiTypography-body1"
      >
        Region:
      </p>
      <p
        class="MuiTypography-root makeStyles-cardText-8 MuiTypography-body1"
      />
    </div>
    <div
      class="MuiBox-root MuiBox-root-14 makeStyles-textBox-5"
    >
      <p
        class="MuiTypography-root makeStyles-cardTitle-7 MuiTypography-body1"
      >
        Capital:
      </p>
      <p
        class="MuiTypography-root makeStyles-cardText-8 MuiTypography-body1"
      />
    </div>
  </div>
</div>
`;
